datafusion/core/src/physical_planner.rs

```diff
@@ -234,7 +234,7 @@ fn create_physical_name(e: &Expr, is_first_expr: bool) -> Result<String> {
                     stride: _,
                 } => {
                     unreachable!(
-                        "ListRange should have been rewritten in OperatorToFunction"
+                        "ListRange should have been rewritten in ArrayFunctionRewriter"
```

Comment:
OperatorToFunction was the old name for these functions I think

Reply:
Actually, now that I think about this I think the issue is that we now expect that these Exprs should never be present....

I think they can still be created via `Expr::field(name)`... 

Will double check / see if I can make a reproduicer

Reply:
Indeed -- it panics here: https://github.com/apache/datafusion/issues/10565